File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        dummyNode = ListNode(-1)
        tailNode = dummyNode
        while True:
            minNodeIdx = -1
            for i in range(len(lists)):
                if not lists[i]:
                    continue
                if minNodeIdx == -1 or lists[i].val < lists[minNodeIdx].val:
                    minNodeIdx = i
            if minNodeIdx == -1:
                break
            else:
                tailNode.next = lists[minNodeIdx]
                tailNode = tailNode.next
                lists[minNodeIdx] = lists[minNodeIdx].next
        return dummyNode.next

            


    
        

File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head == nullptr)
            return nullptr;
        ListNode* nodePrev = head;
        ListNode* nodeNext = head->next;
        nodePrev->next = nullptr;
        while(nodeNext){
            ListNode* temp = nodeNext;
            nodeNext = nodeNext->next;
            temp->next = nodePrev;
            nodePrev = temp;
        }
        return nodePrev;
    }
};
